api/external_auth.proto

```diff
@@ -0,0 +1,150 @@
+syntax = "proto3";
+
+// [#proto-status: draft]
+
+package envoy.api.v2;
```

Comment:
for new stuff I would like to be a bit better about namespacing. Can we move this into `envoy.api.v2.auth` or something like that? And potentially put into an auth directory under api? Since it's not used yet can you also move auth.proto into that new namespace/directory?